extensions/grpc/runtime/src/main/java/io/quarkus/grpc/auth/GrpcSecurityInterceptor.java

```diff
@@ -0,0 +1,122 @@
+package io.quarkus.grpc.auth;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.Prioritized;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jboss.logging.Logger;
+
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.quarkus.grpc.GlobalInterceptor;
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AuthenticationRequest;
+import io.quarkus.security.runtime.SecurityIdentityAssociation;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+/**
+ * Security interceptor invoking {@link GrpcSecurityMechanism} implementations
+ */
+@GlobalInterceptor
+@Singleton
+public final class GrpcSecurityInterceptor implements ServerInterceptor, Prioritized {
+
+    private static final Logger log = Logger.getLogger(GrpcSecurityInterceptor.class);
+
+    private final IdentityProviderManager identityProviderManager;
+    private final SecurityIdentityAssociation identityAssociation;
```

Comment:
This should be able to just be the `io.quarkus.security.identity.CurrentIdentityAssociation` interface instead.